crates/subspace-core-primitives/src/objects.rs

```diff
@@ -126,17 +126,19 @@ pub struct PieceObjectMapping {
 }
 
 /// Object stored in the history of the blockchain
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Encode, Decode, TypeInfo)]
```

Comment:
What was the reason for removing some of the implementations here and below?

Reply:
The automatically-derived Ord impls don’t make sense with a hash as the first field. Instead, we could:
- put the hash last in the struct, so that mappings are ordered by piece index and offset
- write a custom Ord impl to achieve the same thing

I went with the simplest alternative in commit a2ffc5ca42 and documented it in the comments. I also added a missing hash() method.

Reply:
It was less about specific order and more about some order. We have other mappings that use the old layout (though we can't change them right now due to that affecting encoding, maybe have a TODO to change it when we break the protocol compatibility). Explicit implementation of the traits is also an option, but it is fine as it is right now I think.